scripts/irods/execute.py

```diff
@@ -80,51 +78,42 @@ def execute_command_timeout(args, timeout=10, **kwargs):
                 p.kill()
         except OSError:
             pass
-        raise IrodsError(
-            'The call {0} did not complete within'
-            ' {1} seconds.'.format(args, timeout))
+        raise IrodsError('The call {0} did not complete within {1} seconds.'.format(args, timeout))
 
-def execute_command_permissive(args, stdout=subprocess.PIPE, stderr=subprocess.PIPE, input=None, foreground=False, **kwargs):
+def execute_command_permissive(args, stdout=subprocess.PIPE, stderr=subprocess.PIPE, input=None, **kwargs):
     if input is not None:
         if 'stdin' in kwargs and kwargs['stdin'] != subprocess.PIPE:
             raise IrodsError('\'input\' option is mutually exclusive with a \'stdin\' '
                     'option that is not equal to \'subprocess.PIPE\'.')
         kwargs['stdin'] = subprocess.PIPE
     p = execute_command_nonblocking(args, stdout=stdout, stderr=stderr, **kwargs)
+    out, err = communicate_and_log(p, args, input)
+    return (out, err, p.returncode)
 
-    if foreground == True:
-        while True:
-            line = p.stdout.readline()
-            if line == '' and p.poll() is not None:
-                break
-            sys.stdout.write(line)
-            sys.stdout.flush()
-        return (p.stdout, p.stderr, p.returncode)
-    else:
-        out, err = communicate_and_log(p, args, input)
-        return (out, err, p.returncode)
-
-def check_command_return(args, out, err, returncode, **kwargs):
+def check_command_return(args, out, err, returncode, input=None, **kwargs):
     if returncode is not None and returncode != 0:
         if 'env' in kwargs:
             kwargs_without_env = copy.copy(kwargs)
             kwargs_without_env['env'] = 'HIDDEN'
-        else :
+        else:
             kwargs_without_env = kwargs
+        deets = [
```

Comment:
details?

Reply:
We say "opts" for "options" so I figure "deets" for "details" is fair play

Reply:
i'll allow it - don't tell anyone.